monai/transforms/spatial/array.py

```diff
@@ -515,15 +515,18 @@ def __call__(
         sr = len(original_spatial_shape)
         if sr <= 0:
             raise ValueError("data_array must have at least one spatial dimension.")
+        input_affine: NdarrayOrTensor
         affine_: np.ndarray
-        affine_np: np.ndarray
-        if isinstance(data_array, MetaTensor):
-            affine_np, *_ = convert_data_type(data_array.affine, np.ndarray)
-            affine_ = to_affine_nd(sr, affine_np)
+        if affine is not None:
+            warnings.warn("`affine` argument is deprecated, please use the affine of MetaTensor in data_array.")
```

Comment:
Shall we remove the 'affine' arg in this method for v1.0? It's deprecated and will generate the warning message already via 'deprecated_arg'. If we keep it, I hope metatensor.affine would have a higher priority.

Reply:
I think we should remove this arg in 1.0, I add the functionality back here just because it's important argument and may help users smoothly migrate to MetaTensor.
Updated the priority in the latest commit.

Thanks.

Reply:
So this is only temporarily for the users installing from the dev branch before we remove the args in v1.0? I feel this might cause more confusions...

Reply:
Hi @wyli ,

I think it's for the weekly build releases.
If more users face this problem, at least we can suggest them to pip install the weekly build and then may migrate to MetaTensor if they are interested.
What do you think?

Thanks.

Reply:
so why not recommend using the metatensor to store the affine instead of installing the weekly build? the metatensor usage will be the v1.0 way to do it https://github.com/Project-MONAI/MONAI/issues/4791#issuecomment-1198920001

Reply:
OK, let me set up an offiline discussion on Monday.

Thanks.